python3Packages.tensordict: 0.9.1 -> 0.10.0

GaetanLepage opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

Diff: https://github.com/pytorch/tensordict/compare/v0.9.1...v0.10.0
Changelog: https://github.com/pytorch/tensordict/releases/tag/v0.10.0

cc 


- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python3Packages.tensordict: 0.9.1 -> 0.10.0

Diff: https://github.com/pytorch/tensordict/compare/v0.9.1...v0.10.0

Changelog: https://github.com/pytorch/tensordict/releases/tag/v0.10.0

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 441191`
Commit: [`d07dfa15ca836e701273fb2b0eeb779533215002`](https://github.com/NixOS/nixpkgs/commit/d07dfa15ca836e701273fb2b0eeb779533215002) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/d07dfa15ca836e701273fb2b0eeb779533215002..pull/441191/head))
Merge: [`52edef2848a8a4267c92245877ec6798f09791bf`](https://github.com/NixOS/nixpkgs/commit/52edef2848a8a4267c92245877ec6798f09791bf)

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/17550375471


---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.tensordict</summary>
<pre>setting SOURCE_DATE_EPOCH to timestamp 315619200 of file &quot;source/tutorials/src/.gitkeep&quot;
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Executing pypaBuildPhase
Setting SETUPTOOLS_SCM_PRETEND_VERSION to 0.10.0
Including all tracked files automatically
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...

ERROR Missing dependencies:
	pybind11[global]
	pybind11_global==2.13.6</pre>
</details>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.tensordict</summary>
<pre>setting SOURCE_DATE_EPOCH to timestamp 315619200 of file &quot;source/tutorials/src/.gitkeep&quot;
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Executing pypaBuildPhase
Setting SETUPTOOLS_SCM_PRETEND_VERSION to 0.10.0
Including all tracked files automatically
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...

ERROR Missing dependencies:
	pybind11[global]
	pybind11_global==2.13.6</pre>
</details>
</details>


---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.tensordict</summary>
<pre>setting SOURCE_DATE_EPOCH to timestamp 315619200 of file &quot;source/tutorials/src/.gitkeep&quot;
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Executing pypaBuildPhase
Setting SETUPTOOLS_SCM_PRETEND_VERSION to 0.10.0
Including all tracked files automatically
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...

ERROR Missing dependencies:
	pybind11[global]
	pybind11_global==2.13.6</pre>
</details>
</details>

SandaruKasa reviewed (dismissed):
This breaks `python312Packages.tensordict` & `python313Packages.tensordict`

```
┃        > ERROR Missing dependencies:
┃        >      pybind11[global]
┃        >      pybind11_global==2.13.6
```

GaetanLepage committed:
python3Packages.torchrl: 0.9.2 -> 0.10.0

Diff: https://github.com/pytorch/rl/compare/v0.9.2...v0.10.0

Changelog: https://github.com/pytorch/rl/releases/tag/v0.10.0

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 441191`
Commit: `21355e149a87977b72cef2113e718c71351f6096`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

SandaruKasa commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 441191`
Commit: `21355e149a87977b72cef2113e718c71351f6096`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

GaetanLepage commented on pkgs/development/python-modules/torchrl/default.nix:
```diff
@@ -64,33 +73,46 @@
   ];
 
   optional-dependencies = {
     atari = gymnasium.optional-dependencies.atari;
```
Done.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 441191`
Commit: `dfbce9a4d45d67edf7f7add221f6c5133f6b12ea`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>
